contracts/strategies/ELKILPStrategyV5.sol

```diff
@@ -0,0 +1,318 @@
+        devAddr = msg.sender;
+
+        assignSwapPairSafely(_swapPairToken0, _swapPairToken1, _rewardToken);
+        setAllowances();
+        updateMinTokensToReinvest(_minTokensToReinvest);
+        updateAdminFee(_adminFeeBips);
+        updateDevFee(_devFeeBips);
+        updateReinvestReward(_reinvestRewardBips);
+        updateDepositsEnabled(true);
+        transferOwnership(_timelock);
+
+        emit Reinvest(0, 0);
+    }
+
+    /**
+     * @notice Initialization helper for Pair deposit tokens
+     * @dev Checks that selected Pairs are valid for trading reward tokens
+     * @dev Assigns values to swapPairToken0 and swapPairToken1
+     */
+    function assignSwapPairSafely(address _swapPairToken0, address _swapPairToken1, address _rewardToken) private {
+        if (_rewardToken != IPair(address(depositToken)).token0() && _rewardToken != IPair(address(depositToken)).token1()) {
+            // deployment checks for non-pool2
+            require(_swapPairToken0 > address(0), "Swap pair 0 is necessary but not supplied");
+            require(_swapPairToken1 > address(0), "Swap pair 1 is necessary but not supplied");
+            swapPairToken0 = IPair(_swapPairToken0);
+            swapPairToken1 = IPair(_swapPairToken1);
+            require(swapPairToken0.token0() == _rewardToken || swapPairToken0.token1() == _rewardToken, "Swap pair supplied does not have the reward token as one of it's pair");
+            require(
+                swapPairToken0.token0() == IPair(address(depositToken)).token0() || swapPairToken0.token1() == IPair(address(depositToken)).token0(),
+                "Swap pair 0 supplied does not match the pair in question"
+            );
+            require(
+                swapPairToken1.token0() == IPair(address(depositToken)).token1() || swapPairToken1.token1() == IPair(address(depositToken)).token1(),
+                "Swap pair 1 supplied does not match the pair in question"
+            );
+        } else if (_rewardToken == IPair(address(depositToken)).token0()) {
+            swapPairToken1 = IPair(address(depositToken));
+        } else if (_rewardToken == IPair(address(depositToken)).token1()) {
+            swapPairToken0 = IPair(address(depositToken));
+        }
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(stakingContract), MAX_UINT);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
```

Comment:
why onlyAllowedDeposits?

Reply:
Mmmmh I took the code of one of the existing strategy to implement this new one . I think it was `DexStrategyV5.sol` . I can remove this if not needed, I dont know what would be the use case for it in this ELK strategy

Reply:
Yeah looks like it was introduced in the DexV4, @snow-farmer, do you remember why did you add it here?

Reply:
So what should I do regarding this point ?